src/bisection.ts

```diff
@@ -102,11 +114,15 @@ export class ChallengeManager {
       throw new Error('Consensus witness cannot be the last stored state');
     }
 
-    // TODO: With a merkle tree, the witness needs to be validated as opposed to compared to stored states
-    if (this.stateHashes[consensusIndex + 1] !== disputedWitness.witness) {
-      throw new Error('Disputed witness does not match');
+    const validConsensusWitness = validateWitness(consensusWitness, this.root);
+    if (!validConsensusWitness) {
+      throw new Error('Invalid consensus witness proof');
```

Comment:
Didn't see a test case for this. If it doesn't exist, let's add one.

Reply:
Done in 7105426